### PR TITLE
feat(chat-e2e): updated tests with stateful api usage

### PR DESCRIPTION
**Pull request by irinakartun**

### CODE REVIEW

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/README.md [hunk 1]
```diff
@@ -59,11 +59,12 @@ CI report includes screenshots for failed tests.
...
+| `E2E_WORKERS`  | No       | Number of threads to run e2e tests                                                                                                                 | Any number         |                |
+| `E2E_PASSWORD` | No       | A password for e2e authentification                                                                                                                | Any string         |                |
+| `TMS_URL`      | No       | TMS URL                                                                                                                                            | Any string         |                |
+| `ISSUE_URL`    | No       | Issue URL                                                                                                                                          | Any string         |                |
+| `STORAGE_TYPE` | No       | Storage type used for e2e tests data                                                                                                               | api,browserStorage | api            |
```
Comment (mikitabut) [inline apps/chat-e2e/README.md]: Maybe it's possible to read this from already defined variables for chat and use API if not set?

File: apps/chat-e2e/src/testData/conversationHistory/conversationBuilder.ts [hunk 2]
```diff
@@ -10,21 +10,19 @@ import {
...
-  private conversation: Conversation;
+  private conversation: TestConversation;
 
   constructor() {
     this.conversation = {
-      id: constructPath(
-        getRootId({ apiKey: ApiKeys.Conversations }),
-        DEFAULT_CONVERSATION_NAME,
-      ),
+      id: uuidv4(),
```
Comment (mikitabut) [inline apps/chat-e2e/src/testData/conversationHistory/conversationBuilder.ts:25]: Are you sure that creating conversation with uuid is good idea? We are having some logic where we are parsing some data from id

File: apps/chat-e2e/src/ui/pages/dialHomePage.ts [hunk 3]
```diff
@@ -39,4 +43,17 @@ export class DialHomePage extends BasePage {
...
   }
+
+  async reloadPage() {
+    await super.reloadPage();
+    const appContainer = this.getAppContainer();
+    await appContainer
+      .getChatBar()
+      .getChatLoader()
+      .waitForState({ state: 'hidden' });
+    await appContainer
```
Comment (mikitabut) [inline apps/chat-e2e/src/ui/pages/dialHomePage.ts:54]: Is it correct that you are doing this twice?

**Review by mikitabut — DISMISSED** [review]

File: apps/chat-e2e/README.md (hunk 1 above)
Comment (irinakartun) [reply apps/chat-e2e/README.md]: agree, fixed

File: apps/chat-e2e/src/testData/conversationHistory/conversationBuilder.ts (hunk 2 above)
Comment (irinakartun) [reply apps/chat-e2e/src/testData/conversationHistory/conversationBuilder.ts:25]: conversation is not created here, this is just an object. I left here uuid for compatibility with local storage mode. If stateful API is used, conversation id is adjusted according to 'path/model__name' format. See itemApiHelper.ts class

File: apps/chat-e2e/src/ui/pages/dialHomePage.ts (hunk 3 above)
Comment (irinakartun) [reply apps/chat-e2e/src/ui/pages/dialHomePage.ts:54]: this is done for both side panels - chat and promts:

`await appContainer
      .getChatBar()
      .getChatLoader()
      .waitForState({ state: 'hidden' });`

`await appContainer
      .getPromptBar()
      .getChatLoader()
      .waitForState({ state: 'hidden' });`

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-683.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by mikitabut — APPROVED** [review]

**Pull request merged**